apis/core/lib/domain/queries/dimension-mappings.ts

```diff
@@ -90,40 +96,44 @@ interface ImportMappingsFromSharedDimension {
   predicate: NamedNode
   validThrough?: Date
   /**
-   * By default, the function will query the database to find unmapped cube values.
-   * Override this in automatic tests to replace a subselect for example with static VALUES clause
+   * By default, a function will query the database to find unmapped cube values.
+   * Override this in automatic tests to with static values
    */
-  unmappedValuesGraphPatterns?: SparqlTemplateResult | string
+  unmappedValuesOverride?: Literal[]
 }
 
-export async function importMappingsFromSharedDimension({ dimensionMapping, dimension, predicate, unmappedValuesGraphPatterns, validThrough }: ImportMappingsFromSharedDimension, client = parsingClient) {
-  let unmappedValuesSelect: SparqlTemplateResult | string
-  if (unmappedValuesGraphPatterns != null) {
-    unmappedValuesSelect = unmappedValuesGraphPatterns
+export async function importMappingsFromSharedDimension({ dimensionMapping, dimension, predicate, unmappedValuesOverride, validThrough }: ImportMappingsFromSharedDimension, client = streamClient) {
+  let unmappedValues: AsyncIterable<{ value: Literal }> | Iterable<{ value: Literal }>
+  if (unmappedValuesOverride != null) {
+    unmappedValues = unmappedValuesOverride.map(value => ({ value }))
   } else {
     const cubeGraph = await findCubeGraph(dimensionMapping, client)
-    unmappedValuesSelect = sparql`{
-    ${unmappedValuesQuery(cubeGraph, dimensionMapping)}
-  }`
+    unmappedValues = await unmappedValuesFromQuery(cubeGraph, dimensionMapping).execute(client.query)
   }
 
-  const insert = INSERT`
-    GRAPH ${dimensionMapping} {
-      ${dimensionMapping} ${prov.hadDictionaryMember} [
-        a ${prov.KeyEntityPair} ;
-        ${prov.pairKey} ?value ;
-        ${prov.pairEntity} ?term
-      ]
+  const terms = await getSharedDimensionTerms({ dimension, predicate, validThrough })
+  const newMappings = through2.obj(function ({ value }: { value: Literal }, _, next) {
+    const pair = terms.get(value.value)
+    if (!pair) {
+      return next()
     }
-  `.WHERE`
-    ${unmappedValuesSelect}
 
-    ${VALUES(...await getSharedDimensionTerms({ dimension, predicate, validThrough }))}
-
-    FILTER (str(?value) = str(?identifier))
-  `
+    const pointer = clownface({ dataset: $rdf.dataset(), graph: dimensionMapping })
+      .node(dimensionMapping)
+      .addOut(prov.hadDictionaryMember, member => {
+        member
+          .addOut(rdf.type, prov.KeyEntityPair)
+          .addOut(prov.pairKey, pair.key)
+          .addOut(prov.pairEntity, pair.entity)
+      })
+
+    for (const quad of pointer.dataset) {
+      this.push(quad)
+    }
+    return next()
+  })
 
-  await insert.execute(client.query)
+  await client.store.post(Readable.from(unmappedValues).pipe(newMappings))
```

Comment:
This is the most important change. The unmapped values are streamed and transformed in the `through2` stream to new dictionary entries.

This removes the need to serialize the result of `getSharedDimensionTerms` as a gigantic list of `VALUES`